front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleTrainsListV2.tsx

```diff
@@ -51,56 +46,48 @@ const ImportTrainScheduleTrainsListV2 = ({
   );
 
   const { refetch: refetchTimetable } =
-    osrdEditoastApi.endpoints.getTimetableById.useQuerySubscription({ id: timetableId });
-  // const [postPathFinding] = osrdEditoastApi.endpoints.postPathfinding.useMutation();
+    osrdEditoastApi.endpoints.getV2TimetableById.useQuerySubscription({ id: timetableId });
   const [postTrainSchedule] = osrdEditoastApi.endpoints.postV2TrainSchedule.useMutation();
 
-  const [status, setStatus] = useState(initialStatus);
   const dispatch = useAppDispatch();
 
-  async function launchTrainSchedules(payload: TrainScheduleForm) {
-    try {
-      await postTrainSchedule({ body: [payload] }).unwrap();
-    } catch (error) {
-      return false;
-    }
-
-    return true;
-  }
   async function generateV2TrainSchedules() {
-    const payloads = generateV2TrainSchedulesPayload(trainsList, timetableId);
+    const payloads = generateV2TrainSchedulesPayloads(trainsList, timetableId);
 
     const trainsCount = payloads.length;
     let successfulTrainsCount = 0;
+    let errorsNb = 0;
     // eslint-disable-next-line no-restricted-syntax
     for await (const payload of payloads) {
-      const success = await launchTrainSchedules(payload);
-      if (success) {
+      try {
+        await postTrainSchedule({ body: [payload] }).unwrap();
         successfulTrainsCount += 1;
+      } catch (error) {
+        errorsNb += 1;
       }
     }
-    if (successfulTrainsCount > 0) {
+    if (successfulTrainsCount > 0 && errorsNb > 0) {
```

Comment:
In your current code, you test 2 cases:
- at least 1 success and 1 failure
- only success
but you don't handle the case where all the trains fail.
I think you can modify the code like this, to handle these 2 cases:
- at least 1 failure
- else, everything succeeded
```suggestion
    if (errorsNb > 0) {
```